### ISSUE TITLE
Create Zenhub Standalone App

### ISSUE THREAD

**Issue by Groostav** [ZenHubIO/support#1076]
> As a frequent Zenhub user experiencing some trouble with the existing intercept-and-rewrite scheme employed by the Zenhub plugin, I would like a standalone application that is written natively for my platform so that I may avoid the quirks associated with zenhub in the browser.

Hey guys,

My understanding of the strategy that Zenhub employs to get all its fancyness into the github webpage, is that the plugin for your browser intercepts various requests from github.com, and re-writes the DOM in browser to contain the various zenhub elements. This strategy has a number of appealing aspects to it, namely that you don't need to write some kind of wholesale github adapter, but at the cost of performance and seamlessness.

I'm a little annoyed with some of the performance aspects of this flow. There are quirks with zenhub _everywhere_. if you click too quickly on things then page loads turn up out-of-order, often UI elements come in several seconds after they're needed (causing me to wait), and the characteristic URLs of a restful API are destroyed, meaning navigating zenhubs own pages by editing the URL directly is often difficult and tedious.

**This issue is a request to implement a native application** that would side-step these performance quirks by offering a more one-pass rendering approach, either by use of native GUI elements or through _some kind of clever `WebView` rendering trick_ to avoid page rewrites. 

> I say `WebView` since that's [the one I'm familiar with](https://docs.oracle.com/javase/8/javafx/api/javafx/scene/web/WebView.html), but I'd wager good money that a (cross-platform?) WPF component exists and is very similar, in fact I suspect running Snoop on the Slack windows app will get you your answer. 

worth nothing: I'm an old firefox user, but with minimal extensions, and having tried chrome it seems to be in the same performance-boat. I do recognize edge as being faster but I'm not sure that extra speed would translate into any meaningful usability enhancements if/when #822 is implemented.

---

**tl;dr: make native Zenhub app so that it's faster**

---

_A sample use case_

I went to create a new pipeline today and populate it with 30 or so issues, referenced from some notes I'd left on a whiteboard in our office. My flow was, starting from our `/project/issues/` page

1. hit `Ctrl` + `D`
2. edit `/issue/1234` to `/issue/5678`, hit `enter`
3. wait for the page to load, wait for the `pipeline` drop down to appear
4. _wait for the pipeline's history to appear, because the pipeline drop-down is not populated until some time after its added to the page_
5. click the pipeline option
6. select my new pipeline
7. go back to step 1

A significant amount of time is spent in steps 3 through 6. This issue would correct that.

**Comment by azenMatt** [comment]
Hi @Groostav - thank you for taking the time to create this detailed feedback. We hear you and you'l be pleased to know that we are currently working on a web application to offer users the choice of using a dedicated web app or the browser extension (or both!) 

We are tracking this in issue #156, which you can subscribe to for updates. 

also - we aren't quite ready for beta users just yet - but if you would like to participate in this in the near future, please drop me an email - support at zenhub dot com and il add you to the beta users list. It would be great to have your help 😄

**Comment by arena** [comment]
a web app is not the same thing as a native app... this should be reopened.
